addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -16,36 +16,41 @@ options.registry.gallery = options.Class.extend({
      */
     start: function () {
         var self = this;
+        // TODO In master: define distinct classes.
+        // Differentiate both instances of this class.
+        this.hasAddImages = this.$el[0].querySelector('we-button[data-add-images]');
 
-        // Make sure image previews are updated if images are changed
-        this.$target.on('image_changed.gallery', 'img', function (ev) {
-            var $img = $(ev.currentTarget);
-            var index = self.$target.find('.carousel-item.active').index();
-            self.$('.carousel:first li[data-target]:eq(' + index + ')')
-                .css('background-image', 'url(' + $img.attr('src') + ')');
-        });
-
-        // When the snippet is empty, an edition button is the default content
-        // TODO find a nicer way to do that to have editor style
-        this.$target.on('click.gallery', '.o_add_images', function (e) {
-            e.stopImmediatePropagation();
-            self.addImages(false);
-        });
-
-        this.$target.on('dropped.gallery', 'img', function (ev) {
-            self.mode(null, self.getMode());
-            if (!ev.target.height) {
-                $(ev.target).one('load', function () {
-                    setTimeout(function () {
-                        self.trigger_up('cover_update');
+        if (this.hasAddImages) {
+            // Make sure image previews are updated if images are changed
+            this.$target.on('image_changed.gallery', 'img', function (ev) {
+                var $img = $(ev.currentTarget);
+                var index = self.$target.find('.carousel-item.active').index();
+                self.$('.carousel:first li[data-target]:eq(' + index + ')')
+                    .css('background-image', 'url(' + $img.attr('src') + ')');
+            });
+    
+            // When the snippet is empty, an edition button is the default content
+            // TODO find a nicer way to do that to have editor style
+            this.$target.on('click.gallery', '.o_add_images', function (e) {
+                e.stopImmediatePropagation();
+                self.addImages(false);
+            });
+    
+            this.$target.on('dropped.gallery', 'img', function (ev) {
+                self.mode(null, self.getMode());
+                if (!ev.target.height) {
+                    $(ev.target).one('load', function () {
+                        setTimeout(function () {
+                            self.trigger_up('cover_update');
+                        });
                     });
-                });
+                }
+            });
+        } else {
+            const $container = this.$('> .container, > .container-fluid, > .o_container_small');
+            if ($container.find('> *:not(div)').length) {
+                self.mode(null, self.getMode());
             }
```

Comment:
I don't understand why do you need to ignore the event listeners if `this.hasAddImages`. I took your solution but without your changes in start except `this.hasAddImages = this.$el[0].querySelector('we-button[data-add-images]');` and it solve the issue. Why the rest of start changes are needed ?

Also if it is needed, I would keep the diff minimal by doing something like
```js
if (!this.hasAddImages) {
    const $container = this.$('> .container, > .container-fluid, > .o_container_small');
    if ($container.find('> *:not(div)').length) {
        self.mode(null, self.getMode());
    }
    return this._super.apply(this, arguments); 
}
// Make sure image previews are updated if images are changed
this.$target.on('image_changed.gallery', 'img', function (ev) {
...
return this._super.apply(this, arguments); 
```

Reply:
It is not strictly needed to make it work - but not doing it makes those events triggered several times.
Doing it also prepares the split into two distinct classes that will be done in master.

I followed your advice to make the diff smaller.